src/components/Root/Root.tsx

```diff
@@ -45,6 +48,32 @@ const Root: FC = () => {
     registerCustomScreens();
   }, []);
 
+  const userLoading = useAccountStore((s) => s.loading);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const getAuthParams = useCallback(() => {
+    const queryParams = new URLSearchParams(window.location.href.split('#')[1]);
+    const token = queryParams.get('token');
+    const refreshToken = queryParams.get('refresh_token');
+    const expires = queryParams.get('expires');
+    return { token, refreshToken, expires: parseInt(expires ?? '') };
+  }, []);
+
+  useEffect(() => {
+    const getUserInfo = async () => {
+      const auth = getAuthParams();
+      if (!auth.token || !auth.refreshToken || !auth.expires) {
+        return;
+      }
+      InPlayer.Account.setToken(auth.token, auth.refreshToken, auth.expires);
+      await getAccount();
+      navigate(location.pathname);
+    };
+    getUserInfo();
+  }, [getAuthParams, location.pathname, navigate, userLoading]);
```

Comment:
I would move this "bootstrap" logic into the `inplayer.account.service.ts` initialize method:

https://github.com/jwplayer/ott-web-app/blob/develop/src/services/inplayer.account.service.ts#L39

You're now creating a hard dependency on the InPlayer integration in the root file. I also think the `useEffect` will run unnecessarily because it depends on the `location.pathname`, which constantly changes when navigating the app.

Reply:
Yes, very nice! The only feedback I have is that the translation comments are missing. When using variables in `t()` calls, i18next parser wont be able to extract all translation keys when running `$ yarn i18next`. 

You can test this by running the i18next script and observe the changes in the translation files. I suggest to always use the i18next script instead of adding translation keys manually to ensure the structure is always correct and all languages are in sync.

Reply:
Got it, thank you. I wasn't aware of the i18next script and was doing all the translation keys manually, updated in https://github.com/jwplayer/ott-web-app/pull/296/commits/bbc18f267e65a540ff2f3d4cfc3eee8f5be27f90.